### DISCUSSION TITLE
GeoJSON or Polygons

### DISCUSSION THREAD

**Discussion by itayphytech** [react-native-maps/react-native-maps#5129] [Q&A]
What are the performance and usability differences between using GeoJSON and polygons for rendering geospatial data in React Native Maps? Can you provide examples of when to use each approach and any optimization techniques for improving performance?

**Comment by perryfaro** [comment]
When using GeoJSON, it is internally converted back to a standard marker or polygon, for example. This conversion requires additional functions, which may cause a slight delay in rendering, though this will be minimal. The advantage of using GeoJSON is that you only need to write a single implementation, after which you're done.

In our app, we exclusively use GeoJSON and have not encountered any performance issues.

Below is the link to the source code where GeoJSON is converted:

https://github.com/react-native-maps/react-native-maps/blob/master/src/Geojson.tsx
